Source/NamingConventionValidation/Private/EditorNamingValidatorSubsystem.cpp

```diff
@@ -0,0 +1,463 @@
+    slow_task.Visibility = show_if_no_failures ? ESlowTaskVisibility::ForceVisible : ESlowTaskVisibility::Invisible;
+
+    if ( show_if_no_failures )
+    {
+        slow_task.MakeDialogDelayed( 0.1f );
+    }
+
+    FMessageLog data_validation_log( "NamingConventionValidation" );
+
+    auto num_files_checked = 0;
+    auto num_valid_files = 0;
+    auto num_invalid_files = 0;
+    auto num_files_skipped = 0;
+    auto num_files_unable_to_validate = 0;
+
+    const auto num_files_to_validate = asset_data_list.Num();
+
+    for ( const auto & asset_data : asset_data_list )
+    {
+        slow_task.EnterProgressFrame( 1.0f / num_files_to_validate, FText::Format( LOCTEXT( "ValidatingNamingConventionFilename", "Validating Naming Convention {0}" ), FText::FromString( asset_data.GetFullName() ) ) );
+
+        FText error_message;
+        const auto result = IsAssetNamedCorrectly( error_message, asset_data );
+
+        switch ( result )
+        {
+            case ENamingConventionValidationResult::Excluded:
+            {
+                data_validation_log.Info()
+                    ->AddToken( FAssetNameToken::Create( asset_data.PackageName.ToString() ) )
+                    ->AddToken( FTextToken::Create( LOCTEXT( "ExcludedNamingConventionResult", "has not been tested based on the configuration." ) ) )
+                    ->AddToken( FTextToken::Create( error_message ) );
+
+                ++num_files_skipped;
+            }
+            break;
+            case ENamingConventionValidationResult::Valid:
+            {
+                ++num_valid_files;
+                ++num_files_checked;
+            }
+            break;
+            case ENamingConventionValidationResult::Invalid:
+            {
+                data_validation_log.Error()
+                    ->AddToken( FAssetNameToken::Create( asset_data.PackageName.ToString() ) )
+                    ->AddToken( FTextToken::Create( LOCTEXT( "InvalidNamingConventionResult", "does not match naming convention." ) ) )
+                    ->AddToken( FTextToken::Create( error_message ) );
+
+                ++num_invalid_files;
+                ++num_files_checked;
+            }
+            break;
+            case ENamingConventionValidationResult::Unknown:
+            {
+                if ( show_if_no_failures )
+                {
+                    FFormatNamedArguments arguments;
+                    arguments.Add( TEXT( "ClassName" ), FText::FromString( asset_data.AssetClass.ToString() ) );
+
+                    data_validation_log.Warning()
+                        ->AddToken( FAssetNameToken::Create( asset_data.PackageName.ToString() ) )
+                        ->AddToken( FTextToken::Create( LOCTEXT( "UnknownNamingConventionResult", "has no known naming convention." ) ) )
+                        ->AddToken( FTextToken::Create( FText::Format( LOCTEXT( "UnknownClass", " Class = {ClassName}" ), arguments ) ) );

```

Comment:
why using a named argument here and not just {0} like most of the other calls?

Reply:
that's a copy/paste from the DataValidtion code of UE4. But it could have been too